cli/data_formatter.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module provides utilities for formatting text.
+
+The DataFormatter class provides utility functions such as formatting tabular
+data and json data.
+"""
+
+import json
+
+
+class DataFormatter:
+  """Provides utilities for formatting text.
+
+  The DataFormatter class provides utility functions such as formatting tabular
+  data and json data.
+  """
+
+  def build_table(self, headers, values):
+    """Builds a human friendly string of the data in table form.
+
+    Args:
+      headers: Array of strings to be used as the top row of the table,
+        representing the column names.
+      values: Array of tuples containing one row of output (1 value for each
+        column). Each tuple must have the same number of elements, which much
+        also match the length of the headers tuple.
+    """
+    widths = [0] * len(headers)
+
+    for i in range(len(headers)):
+      widths[i] = max(widths[i], len(headers[i]))
+      for j in range(len(values)):
+        widths[i] = max(widths[i], len(values[j][i]))
+
+    separator_row = list(map(lambda w: '-' * w, widths))
+
+    rows = []
+    rows.append(headers)
+    rows.append(separator_row)
+    rows += values
+
+    field_buffer = '  '
+    table = ''
+
+    for i in range(len(rows)):
+      for j in range(len(rows[i])):
+        table += f'{rows[i][j]:{widths[j]}}'
+        table += field_buffer
+      table = table[:-len(field_buffer)]
```

Comment:
field_buffer.join(fields) tends to work well for this.

Reply:
That's much nicer, thanks!